lib/openhab/dsl/items/semantics.rb

```diff
@@ -0,0 +1,179 @@
+      private_constant :SemanticsAction
+
+      # import all the semantics constants
+      [org.openhab.core.semantics.model.point.Points,
+       org.openhab.core.semantics.model.property.Properties,
+       org.openhab.core.semantics.model.equipment.Equipments,
+       org.openhab.core.semantics.model.location.Locations].each do |parent_tag|
+        parent_tag.stream.for_each do |tag|
+          const_set(tag.simple_name.to_sym, tag.ruby_class)
+        end
+      end
+
+      # put ourself into the global namespace, replacing the action
+      ::Semantics = self # rubocop:disable Naming/ConstantName
+
+      # Checks if this Item is a Location
+      # @return [true, false]
+      def location?
+        SemanticsAction.location?(self)
+      end
+
+      # Checks if this Item is an Equipment
+      # @return [true, false]
+      def equipment?
+        SemanticsAction.equipment?(self)
+      end
+
+      # Checks if this Item is a Point
+      # @return [true, false]
+      def point?
+        SemanticsAction.point?(self)
+      end
+
+      # Checks if this Item has any semantic tags
+      # @return [true, false]
+      def semantic?
+        !!semantic_type
+      end
+
+      # Gets related Equipment [GenericItem] an [Item] belongs to.
+      #
+      # If this Item is a Location and a [GroupItem], it will return
+      # an array of all Equipment items within the Location. Otherwise
+      # it returns the single related Equipment Item this Item belongs
+      # to (or none, if it doesn't belong to any).
+      #
+      # Note that an Equipment Item will return itself.
+      #
+      # @return [GenericItem, Array<GenericItem>, nil]
+      def equipment
+        return SemanticsAction.get_equipment(self) unless location? && is_a?(GroupItem)
+
+        members.select(&:equipment?)
+      end
+
+      # Returns the sub-class of [Equipment] this Item is tagged with.
+      # @return [Class]
+      def equipment_type
+        SemanticsAction.get_equipment_type(self)&.ruby_class
+      end
+
+      # Gets the related Location Item of this Item.
+      # @return [GenericItem]
+      def location
+        SemanticsAction.get_location(self)
+      end
+
+      # Returns the sub-class of [Location] this Item is tagged with.
+      # @return [Class]
+      def location_type
+        SemanticsAction.get_location_type(Self)&.ruby_class
+      end
+
+      # Gets the [Location] Items belonging to this Location Item.
+      # @return [Array<GenericItem>]
+      def locations
+        return [] unless location? && is_a?(GroupItem)
+
+        members.select(&:location?)
+      end
+
+      # Returns the sub-class of [Point] this Item is tagged with.
+      # @return [Class]
+      def point_type
+        SemanticsAction.get_point_type(self)&.ruby_class
+      end
+
+      # Gets the [Point] Items belonging to this Equipment Item.
+      # @return [Array<GenericItem>]
+      def points
+        return [] unless equipment? && is_a?(GroupItem)
```

Comment:
Hmm, I didn't think Points were allowed to belong directly to a Location. But I see I was wrong. So yeah, I'll need to rework this a bit.

As for "letting the underlying openhab code", there _isn't_ a method to on the Semantics script action to get all the points that belong to an Equipment or Location.

Reply:
I am actually hoping that they would allow an equipment item to also be a point and have property, as in assign an equipment, a point and a property into the one item. It seems redundant to have to create a Group to become the equipment group for one point. I haven't voiced the request yet, since I've only just started learning about the semantic model.

Reply:
It seems to work for some things, but not a others. It's really quite annoying. Cause yeah having to introduce an intermediate group for the equipment a huge pain when you have hundreds of single-point-equipments. And by "work" I mean "shows up and is usable in the MainUI on any of the locations, equipment, or points tab."